test/model_test.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+require File.expand_path('../helper', __FILE__)
+
+describe 'models' do
+  before do
+    Phenix.rise!(with_schema: true)
+    ActiveRecord::Base.establish_connection(RAILS_ENV.to_sym)
+    require 'models'
+  end
+
+  describe 'primary_key' do
+    it "should be set to 'id' by default" do
+      User.connection.expects(:execute).never
+      assert_equal 'id', User.primary_key
+    end
+
+    it 'should keep value if already set' do
+      class UniqueModel < ActiveRecord::Base
+        self.primary_key = 'model_id'
+      end
+
+      assert 'model_id', UniqueModel.primary_key
+    end
+  end
+end
```

Comment:
They fail for me if I remove the lib changes (the second test will pass as it covers a change I included -- not overwriting primary_key if already set --, so it being green is expected).

Not happy with the test itself, lemme know if you have suggestions, or if the behavior is different for you:

```diff
(2.2.3) vagrant: /harmony/code/active_record_shards (igor/primary_key)$ git diff
diff --git a/lib/active_record_shards/model.rb b/lib/active_record_shards/model.rb
index be85de7..ca697b7 100644
--- a/lib/active_record_shards/model.rb
+++ b/lib/active_record_shards/model.rb
@@ -50,8 +50,8 @@ module ActiveRecordShards
 
     def self.extended(base)
       base.send(:include, InstanceMethods)
-      base.singleton_class.send(:alias_method, :primary_key_without_default_value, :primary_key)
-      base.singleton_class.send(:alias_method, :primary_key, :primary_key_with_default_value)
+      # base.singleton_class.send(:alias_method, :primary_key_without_default_value, :primary_key)
+      # base.singleton_class.send(:alias_method, :primary_key, :primary_key_with_default_value)
       base.after_initialize :initialize_shard_and_slave
     end
 
(2.2.3) vagrant: /harmony/code/active_record_shards (igor/primary_key)$ for i in 4.2 5.0 5.1; do BUNDLE_GEMFILE="gemfiles/rails$i.gemfile" testrbl test/model_test.rb; done
ruby -rbundler/setup ./test/model_test.rb
Run options: --seed 10323

# Running:

.F

Finished in 4.448047s, 0.4496 runs/s, 0.2248 assertions/s.

  1) Failure:
models::primary_key#test_0001_should be set to 'id' by default [/opt/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activerecord-4.2.8/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:316]:
unexpected invocation: #<ActiveRecord::ConnectionAdapters::Mysql2Adapter:0x7f50890401b0>.execute('SHOW TABLES ', 'SCHEMA')
unsatisfied expectations:
- expected never, invoked once: #<ActiveRecord::ConnectionAdapters::Mysql2Adapter:0x7f50890401b0>.execute(any_parameters)


2 runs, 1 assertions, 1 failures, 0 errors, 0 skips
ruby -rbundler/setup ./test/model_test.rb
Run options: --seed 14563

# Running:

.F

Finished in 4.632979s, 0.4317 runs/s, 0.2158 assertions/s.

  1) Failure:
models::primary_key#test_0001_should be set to 'id' by default [/opt/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activerecord-5.0.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:225]:
unexpected invocation: #<ActiveRecord::ConnectionAdapters::Mysql2Adapter:0x7f604b429700>.execute('SELECT table_name FROM information_schema.tables WHERE table_schema = DATABASE()', 'SCHEMA')
unsatisfied expectations:
- expected never, invoked once: #<ActiveRecord::ConnectionAdapters::Mysql2Adapter:0x7f604b429700>.execute(any_parameters)


2 runs, 1 assertions, 1 failures, 0 errors, 0 skips
ruby -rbundler/setup ./test/model_test.rb
Run options: --seed 481

# Running:

.F

Finished in 4.889283s, 0.4091 runs/s, 0.2045 assertions/s.

  1) Failure:
models::primary_key#test_0001_should be set to 'id' by default [/opt/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/activerecord-5.1.2.rc1/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:223]:
unexpected invocation: #<ActiveRecord::ConnectionAdapters::Mysql2Adapter:0x7fb1de1a65c0>.execute('SELECT table_name FROM information_schema.tables WHERE table_schema = database()', 'SCHEMA')
unsatisfied expectations:
- expected never, invoked once: #<ActiveRecord::ConnectionAdapters::Mysql2Adapter:0x7fb1de1a65c0>.execute(any_parameters)
```